src/Illuminate/Support/Number.php

```diff
@@ -427,4 +428,46 @@ protected static function ensureIntlExtensionIsInstalled()
             throw new RuntimeException('The "intl" PHP extension is required to use the ['.$method.'] method.');
         }
     }
+
+    /**
+     * Convert an integer into its Roman numeral representation.
+     *
+     * @param  int  $number  The number to convert (must be between 1 and 3999)
+     * @return string The Roman numeral representation
+     *
+     * @throws InvalidArgumentException If the number is not between 1 and 3999
+     */
+    public static function roman(int $number): string
+    {
+        if ($number <= 0 || $number > 3999) {
+            throw new InvalidArgumentException('Number must be between 1 and 3999.');
+        }
+
+        $map = [
+            'M' => 1000,
+            'CM' => 900,
+            'D' => 500,
+            'CD' => 400,
+            'C' => 100,
+            'XC' => 90,
+            'L' => 50,
+            'XL' => 40,
+            'X' => 10,
+            'IX' => 9,
+            'V' => 5,
+            'IV' => 4,
+            'I' => 1,
+        ];
+
+        $result = '';
+
+        foreach ($map as $roman => $value) {
+            while ($number >= $value) {
+                $result .= $roman;
+                $number -= $value;
+            }
+        }
```

Comment:
You could probably speed this up when possible by providing a shortcut:
```suggestion
        $map = [
            1000 => 'M',
            900 => 'CM',
            500 => 'D',
            400 => 'CD',
            100 => 'C',
            90 => 'XC',
            50 => 'L',
            40 => 'XL',
            10 => 'X',
            9 => 'IX',
            5 => 'V',
            4 => 'IV',
            1 => 'I',
        ];
        
        if (array_key_exists($number, $map)) {
        	return $map[$number];
        }

        $result = '';

        foreach ($map as $value => $roman) {
            while ($number >= $value) {
                $result .= $roman;
                $number -= $value;
            }
        }
```